packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+// Translations Required
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled ("ON").
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label of the Switch component.
+   */
+  label: PropTypes.node.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Parameters: 1. event 2. switch value
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  id,
+  labelId,
+  checked,
+  disabled,
+  onChange,
+  label,
+  ...customProps
+}) => {
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    sliderButtonRef.current.focus();
+    if (onChange) {
+      onChange(event, checked);
```

Comment:
`onChange` should return the updated value.

```jsx
onChange(!checked, event)
```

Reply:
I have this doubt from beginning, As of now we are not doing any state changes within switch component to change the switch value. with prop checked being provided to consumer. consumer should always manipulate value of checked to change the switch state as shown in example.  so that user will always hold the control on switch state. 

```javascript
    <Switch
      id="defaultSwitchOn"
      labelId="label3"
      checked={value}
      label="Label"
      onChange={(checked, event) => { setValue(!checked); }}
    />
```

if we update switch value in onClick of switch regardless of prop-value then it would be an uncontrolled component and user might not be able control the value for checked prop.

So before making this change would like to know is it expected to behave as mentioned above.

Reply:
This change does not require making any internal state changes in the Switch component. Controlled components are expected to return a new value in the onChange function that represents the anticipated next value of the component. 

Here is an example using a native controlled checkbox implementation. The switch component should behave the same way. The onChange function returns the anticipated next value.

```jsx
const Example = () => {
  const [checked, setChecked] = useState(false);

  const handleChange = event => {
    setChecked(event.target.checked);
  };

  return <input type="checkbox" onChange={handleChange} checked={checked} />;
};
```

An uncontrolled implementation of the Switch may be a feature we introduce in the future, but it is not part of this change.

Reply:
Thank you for sharing the information. [updated examples and onChange parameter.](https://github.com/cerner/terra-core/commit/fcbd00f1ab0c8f611b576559dc12aaa9e5bd6173#diff-ce03daf8e4362de3a438b2f322dedf6f)